common/persistence/dataInterfaces.go

```diff
@@ -811,6 +831,7 @@ type (
 		CurrReplicationTasks []Task
 		CurrTransferTasks    []Task
 		CurrTimerTasks       []Task
+		VersionHistories     *workflow.VersionHistories
```

Comment:
there should be 2 version histories
one for current workflow, one for new workflow after reset

Reply:
This one is redundant. The VersionHistories current locates in the execution info which this request already has current one and an insert one.